src/main/java/com/squareup/kotlinpoet/AnnotationSpec.kt

```diff
@@ -171,11 +208,51 @@ class AnnotationSpec private constructor(builder: AnnotationSpec.Builder) {
         = builder.addMember(name, "%T::class", t)
 
     override fun visitArray(values: List<AnnotationValue>, name: String): Builder {
+      var arrayElementsVisitor: ArrayElementsVisitor? = null
       for (value in values) {
-        value.accept(this, name)
+        if (arrayElementsVisitor == null) {
+          arrayElementsVisitor = ArrayElementsVisitor(builder, value.value.toArrayTypeName())
+        }
+        value.accept(arrayElementsVisitor, name)
       }
       return builder
     }
+
+    private fun Any.toArrayTypeName() = when (this) {
+      is Boolean -> BooleanArray::class.asTypeName()
+      is Byte -> ByteArray::class.asTypeName()
+      is Char -> CharArray::class.asTypeName()
+      is Double -> DoubleArray::class.asTypeName()
+      is Float -> FloatArray::class.asTypeName()
+      is Int -> IntArray::class.asTypeName()
+      is Long -> LongArray::class.asTypeName()
+      is Short -> ShortArray::class.asTypeName()
+      else -> kotlin.Array<Any>::class.asTypeName()
```

Comment:
This won't work. You'll get a raw `Array` type not a generic. This has to be a `ParameterizedTypeName(ARRAY, ANY)`.